lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/AvroTypeException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.avro;
+
+/**
+ * Non-runtime version of {@link org.apache.avro.AvroTypeException} that forces calling libraries to handle type issues found during resolution
+ */
+public class AvroTypeException
+        extends Exception
```

Comment:
Any reason to not make this extend IOException?  Generally exceptions in this module extend IOException because reading typically uses libraries that throw IOExceptions

Reply:
This exception should be used for type reconciliation/compatibility errors within the avro library or my processing on top of it. For example an avro file can be successfully initialized ( the schema read from the file header) without IO issues, but then the type compatibility between the file schema and the reader schema provided by the connector clash.
Should only happen at the init of the page reader, while IOExceptions could happen throughout the use. 
Might not be a worthwhile distinction so I can also have it extend IO for easier handling.